File: teckwah-TMS/backend/utils/logger.py
"""
통합 로깅 시스템 (간소화)
"""

import logging
import os
import sys
from datetime import datetime

# 로그 레벨 설정
LOG_LEVEL = os.environ.get("LOG_LEVEL", "INFO").upper()

# 로거 생성
logger = logging.getLogger("teckwah-tms")
logger.setLevel(getattr(logging, LOG_LEVEL))

# 처리기가 없으면 기본 처리기 추가
if not logger.handlers:
    # 콘솔 출력 핸들러
    console_handler = logging.StreamHandler(sys.stdout)
    console_handler.setLevel(getattr(logging, LOG_LEVEL))

    # 로그 포맷 설정
    log_format = "%(asctime)s [%(levelname)s] %(message)s"
    formatter = logging.Formatter(log_format, datefmt="%Y-%m-%d %H:%M:%S")
    console_handler.setFormatter(formatter)

    logger.addHandler(console_handler)

    # 개발 환경에서만 파일 로깅
    if os.environ.get("ENVIRONMENT") != "production":
        try:
            log_dir = "logs"
            os.makedirs(log_dir, exist_ok=True)
            log_file = f"{log_dir}/app_{datetime.now().strftime('%Y%m%d')}.log"

            file_handler = logging.FileHandler(log_file)
            file_handler.setLevel(getattr(logging, LOG_LEVEL))
            file_handler.setFormatter(formatter)

            logger.addHandler(file_handler)
        except Exception as e:
            logger.warning(f"파일 로깅 설정 중 오류 발생: {e}")

# 간편 API 추가
logger.api = lambda method, url, data=None: logger.info(f"API {method} {url}")
logger.db = lambda msg: logger.debug(f"DB: {msg}")
logger.auth = lambda msg: logger.info(f"AUTH: {msg}")
logger.lock = lambda msg: logger.info(f"LOCK: {msg}")
logger.response = lambda url, success=True: logger.info(
    f"RESPONSE {url}: {'성공' if success else '실패'}"
)
logger.service = lambda service, method, params=None: logger.info(
    f"SERVICE {service}.{method}"
)

# 불필요한 로깅 함수 제거 및 필수 함수 유지

File: teckwah-TMS/frontend/src/utils/logger.js
/**
 * 간소화된 로깅 유틸리티
 * 핵심 실행 포인트에서만 로그를 남기도록 수정
 */

// 프로덕션 환경 여부 확인
const isProduction = process.env.NODE_ENV === 'production';

// 간소화된 로깅 유틸리티
const logger = {
  // API 호출 로그 - 핵심 실행 포인트만 기록
  api: (method, url) => {
    // 프로덕션 환경에서도 중요 API 호출은 로깅
    if (url.includes('/auth/login') || url.includes('/auth/logout')) {
      console.log(`[API] ${method} ${url}`);
    } else if (!isProduction) {
      // 개발 환경에서는 기본 API 호출도 로깅 (파라미터 제외)
      console.log(`[API] ${method} ${url}`);
    }
  },
  
  // API 응답 로그 - 핵심 실행 포인트만 기록
  response: (url, success) => {
    // 프로덕션 환경에서는 중요 API 응답이나 실패한 응답만 로깅
    if (url.includes('/auth/') || !success) {
      console.log(`[API] 응답: ${url} - ${success ? '성공' : '실패'}`);
    } else if (!isProduction) {
      // 개발 환경에서는 모든 API 응답 상태 로깅
      console.log(`[API] 응답: ${url} - ${success ? '성공' : '실패'}`);
    }
  },
  
  // 서비스 호출 로그 - 핵심 실행 포인트만 기록
  service: (service, method) => {
    // 프로덕션 환경에서는 인증 관련 서비스만 로깅
    if (service === 'AuthService') {
      console.log(`[SVC] ${service}.${method}() 호출`);
    } else if (!isProduction) {
      // 개발 환경에서는 주요 서비스 호출 로깅
      console.log(`[SVC] ${service}.${method}() 호출`);
    }
  },
  
  // 중요 데이터 검증 로그 - 필수 실행 포인트만 기록
  data: (message) => {
    // 데이터 검증 관련 중요 로그만 유지
    if (message.includes('타입 검증') || message.includes('데이터 변환')) {
      console.log(`[DATA] ${message}`);
    }
  },
  
  // 디버그 로그 - 프로덕션에서는 사용하지 않음
  debug: () => {},
  
  // 일반 로그 - 중요 정보만 기록
  info: (message) => {
    // 핵심 실행 포인트에 관련된 정보만 기록
    if (message.includes('인증') || message.includes('시작') || message.includes('완료')) {
      console.log(message);
    }
  },
  
  // 경고 로그 - 실제 문제가 있는 경우만 기록
  warn: (message) => {
    console.warn(message);
  },
  
  // 오류 로그 - 항상 기록
  error: (message, error) => {
    if (error) {
      console.error(`${message}:`, error);
    } else {
      console.error(message);
    }
  }
};

export default logger;
